CDCgov/prime-reportstream
# Restore the security to the receiver check PR

This PR restores the security to the receiver check PR.

Test Steps:
Test Steps:

Pull down entire branch
Do a Gradle clean and package
Run dev environment
Ensure that there is a receiver configured for REST transport, in this example we use oh-doh
check the receiver using the command line with: curl -X POST -H "client:simple_report" -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
confirm that the command returns JSON with "result": "success"
stop the REST mock container to simulate a disconnected receiver: docker-compose stop rest-webservice
check the receiver again with: curl -X POST -H "client:simple_report" -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
confirm that the command returns JSON with "result": "fail"
Ensure that there is a receiver configured for SFTP transport, example ak-phd
check this receiver with: curl -X POST -H "client:simple_report" -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
confirm that the command returns JSON with "result": "success"
stop the SFTP mock container to simulate a disconnected receiver: docker-compose stop sftp
check this receiver with: curl -X POST -H "client:simple_report" -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
confirm that the command returns JSON with "result": "fail"
run gradle testEnd2End
run integration test for RESTTransport
have a nice day
## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue
- https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/7284

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
